types/fastly:backend.d.ts

```diff
@@ -0,0 +1,93 @@
+declare module 'fastly:backend' {
+  interface BackendConfiguration {
+    /**
+     * The name of the backend.
+     */
+    name: string;
+    /**
+     * A hostname, IPv4, or IPv6 address for the backend as well as an optional port.
+     * E.G. "hostname", "ip address", "hostname:port", or "ip address:port"
+     */
+    target: string;
+    /**
+     * If set, will force the HTTP Host header on connections to this backend to be the supplied value.
+     */
+    hostOverride?: string;
+    /**
+     * Maximum duration in milliseconds to wait for a connection to this backend to be established.
+     * If exceeded, the connection is aborted and a 503 response will be presented instead.
+     * Defaults to 1,000 milliseconds.
+     */
+    connectTimeout?: number;
+    /**
+     * Maximum duration in milliseconds to wait for the server response to begin after a TCP connection is established and the request has been sent.
+     * If exceeded, the connection is aborted and a 503 response will be presented instead.
+     * Defaults to 15,000 milliseconds.
+     */
+    firstByteTimeout?: number;
+    /**
+     * Maximum duration in milliseconds that Fastly will wait while receiving no data on a download from a backend.
+     * If exceeded, the response received so far will be considered complete and the fetch will end.
+     * Defaults to 10,000 milliseconds.
+     */
+    betweenBytesTimeout?: number;
+    /**
+     * Whether or not to require TLS for connections to this backend.
+     */
+    useSSL?: boolean;
+    /**
+     * Minimum allowed TLS version on SSL connections to this backend.
+     * If the backend server is not able to negotiate a connection meeting this constraint, a 503 response will be presented instead.
+     */
+    tlsMinVersion?: number;
```

Comment:
Should we use an enum for `tlsMinVersion` instead of a number? Effectively `"1.1" | "1.2" | "1.3"` as a string?

Reply:
> Should we use an enum for `tlsMinVersion` instead of a number? Effectively `"1.1" | "1.2" | "1.3"` as a string?

Good idea, let's do this in a separate PR to this docs one -- I'm happy to move to some form of enum, what do you think about Symbols instead of strings? Like this -- `import { Backend, TLS_1_2 } from "fastly:backend"`